lib/altsvc.c

```diff
@@ -231,18 +253,40 @@ static CURLcode altsvc_load(struct altsvcinfo *asi, const char *file)
 static CURLcode altsvc_out(struct altsvc *as, FILE *fp)
 {
   struct tm stamp;
+  const char *dst6_pre = "";
+  const char *dst6_post = "";
+  const char *src6_pre = "";
+  const char *src6_post = "";
   CURLcode result = Curl_gmtime(as->expires, &stamp);
   if(result)
     return result;
-
+#ifdef ENABLE_IPV6
+  else {
```

Comment:
I think this would be better if you got rid of the else block and moved the declaration with the rest.

#ifdef ENABLE_IPV6
char ipv6_addr[16];
#endif

Reply:
I sort of like reducing the number of #ifdefs rather than increasing them, which is why I took this approach...

Reply:
Yes that's a benefit. When I read something like  if(foo) return foo; else { code block} I think unnecessary block but either way it's fine.